### PR TITLE
server: refine plan replayer dump file

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: yisaer <disxiaofei@163.com>

### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/26335

Problem Summary:

### What is changed and how it works?

1. Refine play replayer dump File response and log
2. Make plan replayer gc as common function for optimizer trace

### Check List

Tests 

- [x] Unit test

### Release note

```release-note
None
```

### CODE REVIEW

File: domain/plan_replayer.go [hunk 1]
```diff
@@ -56,32 +56,37 @@ func parseTime(s string) (time.Time, error) {
...
+	for _, path := range p.paths {
+		p.gcDumpFilesByPath(path, t)
+	}
+}
+
+func (p *dumpFileGcChecker) gcDumpFilesByPath(path string, t time.Duration) {
 	files, err := ioutil.ReadDir(path)
 	if err != nil {
 		if !os.IsNotExist(err) {
 			logutil.BgLogger().Warn("[PlanReplayer] open plan replayer directory failed", zap.Error(err))
```
Comment (rebelice) [inline domain/plan_replayer.go:71]: [PlanReplayer] -> [gcDumpFilesByPath]

Here we'd better add the `path` in the warn log.

**Review by rebelice — COMMENTED** [review]
We need to change the doc about `plan replayer`.
This PR changes its behavior that it will not remove zip file after dumping any more.
rest LGTM

**Review by rebelice — APPROVED** [review]

**Review by JmPotato — APPROVED** [review]

**Pull request merged**